Improve MongoDB with Panache guide

loicmathieu opened the pull request:
- Update link to the latest driver version
- Improve wording
- Add a section about the PojoCodecProvider

Fixes #16399

gsmet commented on docs/src/main/asciidoc/mongodb-panache.adoc:
```diff
@@ -654,6 +654,21 @@ This can be achieved by setting to DEBUG the following log category inside your
 quarkus.log.category."io.quarkus.mongodb.panache.runtime".level=DEBUG
 ----
 
+== The PojoCodecProvider : easy object to BSON document conversion.
```
```suggestion
== The PojoCodecProvider: easy object to BSON document conversion.
```

gsmet commented on docs/src/main/asciidoc/mongodb-panache.adoc:
```diff
@@ -654,6 +654,21 @@ This can be achieved by setting to DEBUG the following log category inside your
+This codec obeys the Java Bean standard, so it will successfully convert a POJO using public fields or getters/setters.
+You can use `@BsonIgnore` to make a field, or a getter/setter, ignored by the codec.
+
+If you class didn't obey these rules (for example by including a method that starts with `get` but is not a setter),
```
```suggestion
If your class doesn't obey these rules (for example by including a method that starts with `get` but is not a setter),
```

loicmathieu committed:
Improve MongoDB with Panache guide

- Update link to the latest driver version
- Improve wording
- Add a section about the PojoCodecProvider

Fixes #16399

loicmathieu commented:
@gsmet thanks for your review, I applied your suggested changes.

loicmathieu commented:
@gsmet can we merge this one ?

gsmet commented:
Yes :).
